redshift/resource_redshift_default_privileges.go

```diff
@@ -0,0 +1,326 @@
+	if schemaErr != nil {
+		log.Print(schemaErr)
+		tx.Rollback()
+		return schemaErr
+	}
+
+	if isSystemSchema(schemaOwner) {
+		tx.Rollback()
+		return NewError("Privilege creation is not allowed for system schemas, schema=" + schemaName)
+	}
+
+	groupName, groupErr := GetGroupNameForGroupId(tx, d.Get("group_id").(int))
+	if groupErr != nil {
+		log.Print(groupErr)
+		tx.Rollback()
+		return groupErr
+	}
+
+	userName, userErr := getUserFromUserId(tx, d.Get("user_id").(int))
+	if userErr != nil {
+		log.Print(userErr)
+		tx.Rollback()
+		return userErr
+	}
+
+	var revokeStatement = "ALTER DEFAULT PRIVILEGES FOR USER " + userName + " IN SCHEMA " + schemaName + " REVOKE ALL ON TABLES FROM GROUP " + groupName
+
+		if _, err := tx.Exec(revokeStatement); err != nil {
+			log.Print(err)
+			tx.Rollback()
+			return err
+		}
+
+	if len(privilegesSet) > 0 {
+		var grantStatement = "ALTER DEFAULT PRIVILEGES FOR USER " + userName + " IN SCHEMA " + schemaName + " GRANT " + strings.Join(privilegesSet[:], ",") + " ON TABLES TO GROUP " + groupName
+
+		if _, err := tx.Exec(grantStatement); err != nil {
+			log.Print(err)
+			tx.Rollback()
+			return err
+		}
+
+	}
+
+	readErr := readGroupTableDefaultPrivileges(d, tx)
+
+	if readErr != nil {
+		tx.Rollback()
+		return readErr
+	}
+
+	tx.Commit()
+	return nil
+}
+
+func readGroupTableDefaultPrivileges(d *schema.ResourceData, tx *sql.Tx) error {
+	var (
+		selectPrivilege     bool
+		updatePrivilege     bool
+		insertPrivilege     bool
+		deletePrivilege     bool
+		referencesPrivilege bool
+	)
+	var query string
+	query = `
+	      SELECT 
+		decode(charindex('r',split_part(split_part(array_to_string(defaclacl, '|'),'group ' || gr.groname,2 ) ,'/',1)),0,0,1) as select,
+		decode(charindex('w',split_part(split_part(array_to_string(defaclacl, '|'),'group ' || gr.groname,2 ) ,'/',1)),0,0,1) as update,
+		decode(charindex('a',split_part(split_part(array_to_string(defaclacl, '|'),'group ' || gr.groname,2 ) ,'/',1)),0,0,1) as insert,
+		decode(charindex('d',split_part(split_part(array_to_string(defaclacl, '|'),'group ' || gr.groname,2 ) ,'/',1)),0,0,1) as delete,
+		decode(charindex('x',split_part(split_part(array_to_string(defaclacl, '|'),'group ' || gr.groname,2 ) ,'/',1)),0,0,1) as references
+	      FROM pg_group gr, pg_default_acl acl
+	      WHERE 
+		acl.defaclnamespace = $1
+		AND array_to_string(acl.defaclacl, '|') LIKE '%' || 'group ' || gr.groname || '=%'
+		AND gr.grosysid = $2
+		AND acl.defacluser = $3
+		`
+
+	if err := tx.QueryRow(query, d.Get("schema_id").(int), d.Get("group_id").(int), d.Get("user_id").(int)).Scan(
+		&selectPrivilege,
+		&updatePrivilege,
+		&insertPrivilege,
+		&deletePrivilege,
+		&referencesPrivilege); err != nil && err != sql.ErrNoRows {
+		tx.Rollback()
+		return err
+	}
+
+	d.Set("select", selectPrivilege)
+	d.Set("insert", insertPrivilege)
+	d.Set("update", updatePrivilege)
+	d.Set("delete", deletePrivilege)
+	d.Set("references", referencesPrivilege)
```

Comment:
Are we missing usage?

Reply:
usage is a schema permission so doesn't apply to this specific set of permissions